// @redocly/developer-portal/dist/engine/plugins/gatsby-source-redocly-filesystem/rbac/query.js
module.exports=async function(a){const{data:b,errors:c}=await a(`
      query S($ignoreRBAC: Boolean) {
        sidebars: allSidebar {
          nodes {
            name
            permission
            context
            items(ignoreRBAC: $ignoreRBAC) {
              ...DeepNavItems
            }
          }
        }
        siteConfig {
          nav(ignoreRBAC: $ignoreRBAC) {
            type
            label
            link
            external
            target
            activateWithSidebar
            permission
            items(ignoreRBAC: $ignoreRBAC) {
              type
              label
              link
              permission
              items(ignoreRBAC: $ignoreRBAC) {
                type
                label
                link
                permission
              }
            }
          }
          footer {
            copyrightText
            columns(ignoreRBAC: $ignoreRBAC) {
              permission
              group
              items(ignoreRBAC: $ignoreRBAC) {
                ...NavItemDetails
              }
            }
          }
        }
        apis(ignoreRBAC: $ignoreRBAC) {
          id
          definitionId
          link
          sourcePath
          info
          data
          definitionProperties
          permission
        }
        pages: allContentItem {
          nodes {
            link
            permission
          }
        }
      }

      fragment NavItemDetails on NavItem {
        label
        link
        httpVerb
        type
        expanded
        pageId
        redocPrevLink
        redocPrevLabel
        redocNextLink
        redocNextLabel
        redocFakePage
        external
        target
        activateWithSidebar
        menuStyle
        icon
        sublabel
        permission
        separatorLine
        apiVersionId
        isDefaultApiVersion
      }

      fragment DeepNavItems on NavItem {
        ...NavItemDetails
        items(ignoreRBAC: $ignoreRBAC) {
          ...NavItemDetails
          items(ignoreRBAC: $ignoreRBAC) {
            ...NavItemDetails
            items(ignoreRBAC: $ignoreRBAC) {
              ...NavItemDetails
              items(ignoreRBAC: $ignoreRBAC) {
                ...NavItemDetails
                items(ignoreRBAC: $ignoreRBAC) {
                  ...NavItemDetails
                  items(ignoreRBAC: $ignoreRBAC) {
                    ...NavItemDetails
                    items(ignoreRBAC: $ignoreRBAC) {
                      ...NavItemDetails
                    }
                  }
                }
              }
            }
          }
        }
      }
    `,{ignoreRBAC:!0});return console.log(c),c&&process.exit(0),b};
